Mackbellemore/soen-390-team07
# feat(client/server): user Roles

#

## Changes this PR introduces

- User roles and permissions
- 4 types of users, manufacturing, general, admin, finance
- Each type can only view and do certain things


## How to test this PR

- Test every endpoint that I updated with different roles, make sure the endpoints work only with the allowed roles
- Test front end for each role make sure you can only access allowed pages 
- Make sure the logic makes sense, if you think an endpoint is missing a role, comment 







## Notes for reviewers

- Test it properly plz 



## Checklist
